File: Python/PyLife/polygon.py
class Polygon:
    def __init__(self, *points):
        self._points = (*points,)

    def __iter__(self):
        return self._points.__iter__()

    @property
    def points(self) -> tuple:
        return self._points

    def clone(self):
        return Polygon(*self._points)


SQUARE = Polygon(
    (-1.0, -1.0), (1.0, -1.0), (1.0, 1.0), (-1.0, 1.0),
)
DIAMOND = Polygon(
    (0.0, -1.0), (1.0, 0.0), (0.0, 1.0), (-1.0, 0.0),
)
TRIANGLE_N = Polygon(
    (0.0, -1.0), (1.0, 1.0), (-1.0, 1.0),
)
TRIANGLE_E = Polygon(
    (1.0, 0.0), (-1.0, 1.0), (-1.0, -1.0),
)
TRIANGLE_S = Polygon(
    (0.0, 1.0), (-1.0, -1.0), (1.0, -1.0),
)
TRIANGLE_W = Polygon(
    (-1.0, 0.0), (1.0, -1.0), (1.0, 1.0),
)

File: JavaScript/JSLife/polygon.js
class Polygon {
  constructor(/* points */) {
    this._points = [...arguments];
  }
  
  get points() {
    return this._points;
  }
  
  set points(rhs) {
    this._points = rhs;
  }
  
  appendPoint(rhs) {
    this._points.push(rhs);
  }
  
  prependPoint(rhs) {
    this._points.unshift(rhs);
  }
  
  clone() {
    return new Polygon(...this._points);
  }
}

const SQUARE = new Polygon(
    [-1.0, -1.0], [1.0, -1.0], [1.0, 1.0], [-1.0, 1.0],
);
const DIAMOND = new Polygon(
    [0.0, -1.0], [1.0, 0.0], [0.0, 1.0], [-1.0, 0.0],
);
const TRIANGLE_N = new Polygon(
    [0.0, -1.0], [1.0, 1.0], [-1.0, 1.0],
);
const TRIANGLE_E = new Polygon(
    [1.0, 0.0], [-1.0, 1.0], [-1.0, -1.0],
);
const TRIANGLE_S = new Polygon(
    [0.0, 1.0], [-1.0, -1.0], [1.0, -1.0],
);
const TRIANGLE_W = new Polygon(
    [-1.0, 0.0], [1.0, -1.0], [1.0, 1.0],
);
